pandas/core/frame.py

```diff
@@ -5199,6 +5199,34 @@ def drop_duplicates(
         else:
             return result
 
+    def is_unique(
+        self, subset: Optional[Union[Hashable, Sequence[Hashable]]] = None
+    ) -> Series:
+        """
+        Return boolean Series denoting columns with unique values.
+
+        Parameters
+        ----------
+        subset : column label or sequence of labels, optional
+            Only consider certain columns for finding uniques. by default use columns.
+
+        Returns
+        -------
+        Series
+
+        See Also
+        --------
+        DataFrame.duplicated : Indicate duplicate rows.
+        """
+        if subset is not None:
+            subset = subset if is_list_like(subset) else [subset]
+            self = self[subset]
+
+        if len(self.columns):
+            return self.apply(lambda x: x.is_unique)
```

Comment:
why does this not work on empties?

Reply:
A dataframe with no columns doesn't call the inner func in apply (here `lambda x: x.is_unique`), so the output (empty) series has to guess its own dtype. It guesses `float64`, which is wrong, so we have to special-case it.